nservicebus/ravendb/unit-of-work.md

```diff
@@ -14,31 +14,55 @@ Share the session between the message handler(s) and the actual unit of work imp
 
 Instead, beginning with NServiceBus version 3, use the new [support for child containers](/nservicebus/containers/child-containers.md). This means that all dependencies configured as a single call effectively become static within the context of one transport message.
 
-To resolve a RavenDB document session from the container, add the following configuration (StructureMap is used but any of the other containers except Spring and Unity would work):
-
+To resolve a RavenDB document session from the container, add the following configuration (StructureMap is used but any of the other containers except Spring and Unity would work):  
 
 ```C#
 var store = new DocumentStore {Url = "http://localhost:8080", DefaultDatabase = "MyDatabase"};
 store.Initialize();
  
-ObjectFactory.Configure(c =>
-    {
-        c.For<IDocumentStore>()
-            .Singleton()
-            .Use(store);
-        c.For<IDocumentSession>()
-            .Use(ctx => ctx.GetInstance<IDocumentStore>()
-                        .OpenSession());
-        c.For<IManageUnitsOfWork>()
-            .Use<RavenUnitOfWork>();
-    });
- 
-Configure.With()
-            .StructureMapBuilder(ObjectFactory.Container);
+//ObjectFactory.Configure(c =>  // objectFactory has been deprecated in structureMap
+Container container = new Container( c => 
+                    {
+                        c.For<IDocumentStore>().Singleton()
+                            .Use(dbStore);
+                        c.For<IDocumentSession>()
+                            .Use(ctx => ctx.GetInstance<IDocumentStore>()
+                                        .OpenSession());
+                        c.For<IManageUnitsOfWork>()
+                            .Use<RavenUnitOfWork>();
+                    });
+configuration.UseContainer<StructureMapBuilder>(c =>
+                       c.ExistingContainer(container));
+
+// old version: 
```

Comment:
No, I'm not frustrated. I just told about the problem when old sample code showing how to use the Unit-of-Work was outdated. Its not StructureMap code because they have no interest in an NServiceBus Unit-of-Work in an NServiceBus handler (that has to do with RavenDB, which is also a good choice of example)

So no. This documentation should be simply corrected, and also made simple to understand (I think that is what I did). As you requested, I removed all the old code instead of remarking it. 

It is an important peace of code and important documentation for NServiceBus. 

It uses the built in NServiceBus container builder class, and I wish I knew if there's a way to set the "internal" default container type to  use this container. 

The only structuremap directive is in the last line...   

I tried it and it worked. I would spend some time testing with the other containers, but don't have time for this.